src/context/DIDContext.tsx

```diff
@@ -0,0 +1,107 @@
+;import { EthereumAuthProvider, useViewerConnection, ViewerConnectionState } from "@self.id/framework";
+import { useEffect, createContext, useState } from "react";
+import { ModelTypes } from "@/interfaces";
+import { useWalletAccount } from "@/hooks/useWalletAccount";
+import { useStateMySelfID } from "@/recoilstate/ceramic";
+import { useDID } from "@/recoilstate";
+
+export interface UserContextState {
+    loggedIn: boolean;
+    connection: ViewerConnectionState<ModelTypes> | undefined
+    account: string | undefined;
+    did: string | undefined
+    chainId: number | undefined
+  }
+  
+  const startingState: UserContextState = {
+    loggedIn: false,
+    connection: undefined,
+    account: undefined,
+    did: undefined,
+    chainId: undefined
+  };
+
+export const DIDContext = createContext(startingState);
```

Comment:
[IMO] I would like to unify global state management with recoil, but is there any reason to use context?
 It's working, so I don't think we need to fix it this time.

Reply:
That's because, In order to use SelfId and web3React together, I wanted to have an environment where `connectWallet` and `connectDID` work in a global state.
I had a problem that the login process was invoked multiple times when trying to login in a hook using recoil.